app/com/linkedin/drelephant/spark/fetchers/SparkRestClient.scala

```diff
@@ -100,6 +111,39 @@ class SparkRestClient(sparkConf: SparkConf) {
     }
   }
 
+  private def getLogData(attemptTarget: WebTarget): Option[SparkLogDerivedData] = {
+    val target = attemptTarget.path("logs")
+    logger.info(s"calling REST API at ${target.getUri} to get eventlogs")
+
+    // The logs are stored in a ZIP archive with a single entry.
+    // It should be named as "$logPrefix.$archiveExtension", but
+    // we trust Spark to get it right.
+    resource.managed { getApplicationLogs(target) }.acquireAndGet { zis =>
+      val entry = zis.getNextEntry
+      if (entry == null) {
+        logger.warn(s"failed to resolve log for ${target.getUri}")
+        None
+      } else {
+        val codec = SparkLogClient.compressionCodecForLogName(sparkConf, entry.getName)
+        Some(SparkLogClient.findDerivedData(
```

Comment:
Unfortunately, I don't have access to the production history server logs and I failed to reproduce the issue locally. Your suspicion though sounds convincing.